### PR TITLE
Scheduling/bugfix: Store the path for a transformation source (instead of the actual file)

### PR DESCRIPTION
**Pull request by aolfat**
# Description

Previously transformation sources were stored as: x/y/2022-08-92/z.parquet
This resulted in never being able to fetch the latest source for scheduling.

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: coordinator/Dockerfile.old [hunk 1]
```diff
@@ -24,16 +29,17 @@ COPY ./kubernetes ./kubernetes
 COPY ./types ./types
 COPY ./coordinator/main/main.go ./coordinator/main/main.go
 
+# Building Go app
 # CGD_ENABLED=0 is used to avoid the libc differences in alpine and ubuntu
 ENV CGO_ENABLED=0
-RUN go build ./coordinator/main/main.go
+RUN go install github.com/go-delve/delve/cmd/dlv@latest
```
Comment (sdreyer) [inline coordinator/Dockerfile.old:35]: We probably dont want delve in here right?

File: provider/spark.go [hunk 2]
```diff
@@ -1461,13 +1461,13 @@ func (d *DatabricksExecutor) GetDFArgs(outputURI string, code string, sources []
 func (spark *SparkOfflineStore) GetTransformationTable(id ResourceID) (TransformationTable, error) {
 	spark.Logger.Debugw("Getting transformation table", "ResourceID", id)
 	transformationPath := spark.Store.PathWithPrefix(fileStoreResourcePath(id), false)
-	transformationExactPath, err := spark.Store.NewestFileOfType(transformationPath, Parquet)
+	//transformationExactPath, err := spark.Store.NewestFileOfType(transformationPath, Parquet)
 
-	if err != nil || transformationExactPath == "" {
-		return nil, fmt.Errorf("could not get transformation table at %s: %v", transformationPath, err)
-	}
+	//if err != nil || transformationExactPath == "" {
```
Comment (sdreyer) [inline provider/spark.go:1466]: Remove comment

File: runner/Dockerfile [hunk 3]
```diff
@@ -29,13 +29,16 @@ COPY metadata/search/* ./metadata/search/
 COPY ./provider/ ./provider/
 COPY runner/worker/main/main.go ./runner/worker/main/main.go
 
+
 # CGD_ENABLED=0 is used to avoid the libc differences in alpine and ubuntu
 ENV CGO_ENABLED=0
-RUN go build -o worker ./runner/worker/main
+RUN go install github.com/go-delve/delve/cmd/dlv@latest
+RUN go build -gcflags="all=-N -l" -o worker ./runner/worker/main
```
Comment (sdreyer) [inline runner/Dockerfile:36]: Probably dont want delve in non dev environments

File: provider/spark.go [hunk 4]
```diff
@@ -1459,15 +1461,9 @@ func (d *DatabricksExecutor) GetDFArgs(outputURI string, code string, sources []
 }
 
 func (spark *SparkOfflineStore) GetTransformationTable(id ResourceID) (TransformationTable, error) {
-	spark.Logger.Debugw("Getting transformation table", "ResourceID", id)
 	transformationPath := spark.Store.PathWithPrefix(fileStoreResourcePath(id), false)
-	transformationExactPath, err := spark.Store.NewestFileOfType(transformationPath, Parquet)
```
Comment (sdreyer) [inline provider/spark.go:1464]: Are we not getting the newest file anymore?

Comment (aolfat) [reply provider/spark.go:1464]: I think this function is a bit confusing and I had the same thought when first viewing it.  This function is what's responsible for fetching a TF's source (at least in the offline store)

In the coordinator, it's used to get the source of a transformation when registering a label/feature. Its output is what you see when you go to the .txt file for a label or feature. By omitting the latest file lookup here and delegating it over to feature/training set evaluation (below) we ensure the latest is fetched during evaluation.

I thought about renaming it to getTransformationSource (but I don't think that's exactly how it's used for the SQL online store and I'm punting doing any refactoring rn)

**Pull request merged**